src/armature/Node.ts

```diff
@@ -189,3 +214,83 @@ export class Node {
         return { ...Node.bone, transform: transformationMatrix, isShadeless: true };
     }
 }
+
+/**
+ * A derived `Node` with an additional `geometry` property.
+ */
+export class GeometryNode extends Node {
```

Comment:
Why did you move the `GeometryNode` to the `Node` file?

Reply:
Circular dependency issues. I'm hoping there's a better way but I haven't found one yet. Basically lack of forward declarations is currently causing this

Reply:
Okay, that's fair then 👍

Reply:
I'm going to see if I can use an interface to wrap it. It's kind of messy but maybe less so than the current implementation. Will report back